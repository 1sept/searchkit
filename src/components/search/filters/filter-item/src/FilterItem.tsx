import * as React from "react";

import {
	SearchkitComponent,
	FacetAccessor,
	FastClick,
	SearchkitComponentProps,
	ReactComponentType
} from "../../../../../core"

export interface FilterItemComponentProps {
	bemBlocks:{container:any, option:any},
	toggleFilter:Function,
	resetFilters:Function,
	translate:Function,
  selected:Boolean,
  label:string,
  count: number
}

export class FilterItemComponent extends React.Component<FilterItemComponentProps, any>{
	render(){
		const {bemBlocks, toggleFilter, translate, resetFilters, selected, label, count} = this.props
    const className = bemBlocks.option()
											.state({selected})
											.mix(bemBlocks.container("item"))

		return (
      <FastClick handler={toggleFilter} key={label}>
				<div className={className}>
					<div className={bemBlocks.option("text")}>{label}</div>
					<div className={bemBlocks.option("count")}>{count}</div>
				</div>
			</FastClick>
		)
	}
}
