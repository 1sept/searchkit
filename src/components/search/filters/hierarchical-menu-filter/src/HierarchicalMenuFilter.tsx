import * as React from "react";
import * as _ from "lodash";
import "../styles/index.scss";

import {
	SearchkitComponent,
  HierarchicalFacetAccessor,
	FastClick,
	SearchkitComponentProps
} from "../../../../../core"

export interface HierarchicalMenuFilterProps extends SearchkitComponentProps{
	id:string
	fields:Array<string>
	title:string
}

export class HierarchicalMenuFilter extends SearchkitComponent<HierarchicalMenuFilterProps, any> {
	public accessor:HierarchicalFacetAccessor

	static propTypes = _.defaults({
		id:React.PropTypes.string.isRequired,
		fields:React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
		title:React.PropTypes.string.isRequired
	}, SearchkitComponent.propTypes)

	constructor(props:HierarchicalMenuFilterProps) {
		super(props)
	}

	defineBEMBlocks() {
		var blockClass = this.props.mod || "hierarchical-menu";
		return {
			container:`${blockClass}-list`,
			option:`${blockClass}-option`
		};
	}

	defineAccessor() {
		return new HierarchicalFacetAccessor(
			this.props.id,
			{id:this.props.id, title:this.props.title, fields:this.props.fields, size:0}
		)
	}

	addFilter(option, level) {
		this.accessor.state = this.accessor.state.toggleLevel(
			level,option.key)

		this.searchkit.performSearch()
	}

	renderOption(level, option) {

		var block = this.bemBlocks.option

		var className = block().state({
			selected:this.accessor.state.contains(level, option.key)
		})

		return (
			<div key={option.key}>
				<FastClick handler={this.addFilter.bind(this, option,level)}>
					<div className={className}>
						<div className={block("text")}>{this.translate(option.key)}</div>
						<div className={block("count")}>{option.doc_count}</div>
					</div>
				</FastClick>
					{(() => {
						if(this.accessor.resultsState.contains(level,option.key)) {
							return this.renderOptions(level+1);
						}
					})()}
			</div>
		)
	}

	renderOptions(level) {
		let block = this.bemBlocks.container;
		return (
			<div className={block("hierarchical-options")}>
			{_.map(this.accessor.getBuckets(level), this.renderOption.bind(this,level))}
			</div>
		)
	}

  render(){
		let block = this.bemBlocks.container;
    return (
			<div className={block().mix(`filter--${this.props.id}`)}>
				<div className={block("header")}>{this.props.title}</div>
				<div className={block("root")}>
					{this.renderOptions(0)}
				</div>
			</div>
		)
	}

}
