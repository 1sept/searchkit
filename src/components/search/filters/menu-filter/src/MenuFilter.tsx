import * as React from "react";
import * as _ from "lodash";
import * as classNames from 'classnames';
import {
	Searcher,
	SearchkitManager,
	SearchkitComponent,
	FacetAccessor
} from "../../../../../core"

require("./../styles/index.scss");

export interface IMenuFilter {
	field:string
	title:string
	id:string
}

export class MenuFilter extends SearchkitComponent<IMenuFilter, any> {
	accessor:FacetAccessor

	shouldCreateNewSearcher() {
		return true;
	}

	defineAccessor() {
		return new FacetAccessor(
			this.props.field,
			{id:this.props.id, operator:"OR", title:this.props.title}
		)
	}

	addFilter(option) {
		this.accessor.state = this.accessor.state.clear();
		if (option != "all") {
			this.accessor.state = this.accessor.state.add(option.key);
		}
		this.searchkit.performSearch()
	}

	renderOption(option) {

		let optionClassName = classNames({
			"menu-list-options__item":true,
			"menu-list-option":true,
			"menu-list-option--checked":this.accessor.state.contains(option.key)
		})

		return (
			<div className={optionClassName} key={option.key} onClick={this.addFilter.bind(this, option)}>
				<div className="menu-list-option__text">{option.key}</div>
			</div>
		)
	}

	renderAllOption() {
		let optionClassName = classNames({
			"menu-list-options__item":true,
			"menu-list-option":true,
			"menu-list-option--checked":!this.accessor.resultsState.getValue()
		})

		return (
			<div className={optionClassName} key="all" onClick={this.addFilter.bind(this, "all")}>
				<div className="menu-list-option__text">All</div>
			</div>
		)
	}

	render() {
		var className = classNames({
			"menu-list":true,
			[`filter--${this.props.id}`]:true
		})
		return (
			<div className={className}>
				<div className="menu-list-options">
				{this.renderAllOption()}
				{_.map(this.accessor.getBuckets(), this.renderOption.bind(this))}
				</div>
			</div>
		);
	}
}
