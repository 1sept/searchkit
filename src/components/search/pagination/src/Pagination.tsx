import * as React from "react";
import * as _ from "lodash";
import "../styles/index.scss";

import {
	SearchkitComponent,
	PaginationAccessor,
	FastClick,
	SearchkitComponentProps
} from "../../../../core"

export interface PaginationProps extends SearchkitComponentProps {
}

export enum DIRECTION {
	NEXT,
	PREVIOUS
}

export class Pagination extends SearchkitComponent<PaginationProps, any> {
	accessor:PaginationAccessor

	translations = {
		"pagination.previous":"Previous",
		"pagination.next":"Next"
	}

	static propTypes = _.defaults({
	}, SearchkitComponent.propTypes)

	defineAccessor() {
    return new PaginationAccessor("p")
	}

	defineBEMBlocks() {
		let block = (this.props.mod || "pagination-navigation")
		return {
			container: block,
			option: `${block}-item`
		}
	}

  hasPagination():boolean {
		return true
  }

	getCurrentPage():number {
		return Number(this.accessor.state.getValue()) || 1;
	}

	setPage(direction:DIRECTION) {
		if (this.isDisabled(direction)) { return };
		let currentPage:number = this.getCurrentPage();
		if (direction == DIRECTION.PREVIOUS) {
			this.accessor.state = this.accessor.state.setValue(currentPage-1);
		} else if (direction == DIRECTION.NEXT) {
			this.accessor.state = this.accessor.state.setValue(currentPage+1);
		}
		this.searchkit.performSearch();
		window.scrollTo(0,0);
	}

	isDisabled(direction:DIRECTION):boolean {
		let currentPage:number = this.getCurrentPage();
		let totalPages:number = Math.ceil(
			_.get(this.getResults(),".hits.total",1)
			/
			_.get(this.getQuery(), "query.size", 10)
		)
		if (direction == DIRECTION.PREVIOUS && currentPage == 1) { return true; }
		if (direction == DIRECTION.NEXT && currentPage == totalPages ) { return true; }
		return false;
	}

	paginationElement(direction:DIRECTION, cssClass:string, displayText:string ) {
		let className = this.bemBlocks.option()
			.mix(this.bemBlocks.container("item"))
			.state({
				disabled:this.isDisabled(direction)
			})
    return (
			<FastClick handler={this.setPage.bind(this,direction)}>
	      <div className={className} data-qa={direction}>
	        <div className={this.bemBlocks.option("text")}>{this.translate(displayText)}</div>
	      </div>
			</FastClick>
		)
	}

  render() {
		if(this.hasHits()){
			return (
				<div className={this.bemBlocks.container()} data-qa="pagination">
					{this.paginationElement(DIRECTION.PREVIOUS, "prev", "pagination.previous")}
					{this.paginationElement(DIRECTION.NEXT, "next", "pagination.next")}
      	</div>
			)
		}
		return null
  }
}
