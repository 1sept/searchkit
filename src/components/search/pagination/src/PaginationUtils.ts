
const defaults = require('lodash/defaults')

export class PaginationHelper {
  
  currentPage: number
  totalPages: number
  translate: Function
  pages: Array<any>
  lastPage: number
  
  constructor({currentPage, totalPages, translate}){
    this.currentPage = currentPage
    this.totalPages = totalPages
    this.translate = translate
    this.pages = []
    this.lastPage = 0 // Last added page number
  }
  
  push(item){
    this.pages.push(item)
  }
  
  previous(options={}){
    this.push(defaults({
      key: "previous",
      label: this.translate('pagination.previous'),
      page: this.currentPage > 1 ? (this.currentPage - 1) : undefined,
      disabled: this.currentPage === 1
    }, options))
  }
  
  next(options={}){
    this.push(defaults({
      key: "next",
      label: this.translate('pagination.next'),
      page: this.currentPage < this.totalPages - 1 ? (this.currentPage + 1) : undefined,
      disabled: this.currentPage === this.totalPages
    }, options))
  }
  
  page(pageNumber, options={}){
    if (pageNumber > 0 && pageNumber <= this.totalPages){
      this.push(defaults({
        key: pageNumber,
        label: '' + pageNumber,
        page: pageNumber,
        active: pageNumber == this.currentPage
      }, options))
    }
  }
  
  range(minPage, maxPage, options={}){
    const min = Math.max(1, minPage)
    const max = Math.min(maxPage, this.totalPages)
    for(var i=min; i<= max; i++){
      this.page(i, options)
    }
  }
  
  ellipsis(options={}){
    this.push(defaults({ 
      key: "ellipsis-" + this.pages.length, 
      label: '...', 
      disabled: true 
    }, options))
  }
}