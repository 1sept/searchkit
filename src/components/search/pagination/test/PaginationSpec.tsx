import * as React from "react";
import {mount} from "enzyme";
import {Pagination, PaginationSelect} from "../src/Pagination";
import {SearchkitManager, ImmutableQuery} from "../../../../core";
import {Select} from "../../../ui";
import * as ReactTestRenderer from 'react-test-renderer'
import {
  fastClick, hasClass, jsxToHTML, printPrettyHtml
} from "../../../__test__/TestHelpers"
import * as sinon from "sinon";

describe("Pagination tests", () => {

  beforeEach(() => {

    this.searchkit = SearchkitManager.mock()
    this.searchkit.addDefaultQuery((query)=> {
      return query.setSize(10)
    })

    this.createEnzymeWrapper = (showNumbers=true, pageScope=3, props={}) => {
      this.wrapper = mount(
        <Pagination searchkit={this.searchkit}
                    showNumbers={showNumbers}
                    pageScope={pageScope}
                    {...props}
                    translations={{ "pagination.previous": "Previous Page" }} />
      );
      this.accessor = this.searchkit.accessors.statefulAccessors["p"]
    }

    this.createWrapper = ({page=null, withPages=true, pageScope=3, props={} }={}) => {

      const c = (<Pagination
                  searchkit={this.searchkit}
                  showNumbers={withPages}
                  pageScope={pageScope}
                  {...props}
                  translations={{ "pagination.previous": "Previous Page" }} />
                )

      if (page) {
        this.wrapper = ReactTestRenderer.create(c)
        this.accessor = this.searchkit.accessors.statefulAccessors["p"]
        this.accessor.state = this.accessor.state.setValue(page)
      } else {
        this.accessor = this.searchkit.accessors.statefulAccessors["p"]
      }
      this.wrapper = ReactTestRenderer.create(c)
    }

    this.searchkit.query = new ImmutableQuery().setSize(10)


    this.searchkit.setResults({
      hits:{
        total:80
      }
    })


  });

  describe("rendering", () => {

    it("renders text", () => {
      this.createWrapper({})
      expect(this.wrapper).toMatchSnapshot()
    })

    it("renders without pages", () => {
      this.createWrapper({ withPages: false })
      expect(this.wrapper).toMatchSnapshot()
    })

    it('renders first page options', () => {
      this.createWrapper({})
      expect(this.wrapper).toMatchSnapshot()
    })

    it('renders second page options', () => {
      this.createWrapper({ page: 2 })
      expect(this.wrapper).toMatchSnapshot()
    })

    it('renders eighth page options', () => {
      this.createWrapper({ page: 8 })
      expect(this.wrapper).toMatchSnapshot()
    })

    it("handles showNumbers prop", () => {
      this.createWrapper({ withPages: false })
      expect(this.wrapper).toMatchSnapshot()
    })

    it("handles pageScope prop", () => {
      this.createWrapper({ withPages: true, pageScope: 1 })
      expect(this.wrapper).toMatchSnapshot()
    })

    it("renders no pagination on no results", () => {
      this.searchkit.setResults({hits:{total:0}})
      this.createWrapper({})
      expect(this.wrapper).toMatchSnapshot()
    })

    it("both disabled on only one total page", () => {
      this.searchkit.setResults({ hits: { total: 10 } })
      this.createWrapper()
      expect(this.wrapper).toMatchSnapshot()
    })

  });

  describe("interacting", () => {

    it("interact prev disabled", () => {
      this.createEnzymeWrapper()
      this.accessor.state = this.accessor.state.setValue(1)

      fastClick(this.wrapper.find(".sk-toggle__item").first())
      expect(this.accessor.state.getValue()).toBe(1)
    });

    it("click previous, next", ()=> {
      this.createEnzymeWrapper()
      this.accessor.state = this.accessor.state.setValue(3)
      this.wrapper.update()
      fastClick(this.wrapper.find(".sk-toggle__item").first())
      expect(this.accessor.state.getValue()).toBe(2)
      fastClick(this.wrapper.find(".sk-toggle__item").last())
      fastClick(this.wrapper.find(".sk-toggle__item").last())
      expect(this.accessor.state.getValue()).toBe(4)
    })

    it("ability to click last page", ()=> {
      this.createEnzymeWrapper()
      this.accessor.state = this.accessor.state.setValue(7)
      this.wrapper.update()
      fastClick(this.wrapper.find(".sk-toggle__item").last())
      expect(this.accessor.state.getValue()).toBe(8)
      fastClick(this.wrapper.find(".sk-toggle__item").last())
      expect(this.accessor.state.getValue()).toBe(8)
    })

    it("dividers should not alter state", ()=> {
      this.createEnzymeWrapper()
      this.accessor.state = this.accessor.state.setValue(2)
      this.wrapper.update()
      fastClick(this.wrapper.find("[data-key='ellipsis-6']"))
      //this was NaN before bug fix
      expect(this.accessor.state.getValue()).toBe(2)
    })

  })

  it("PaginationSelect", () => {
    this.wrapper = ReactTestRenderer.create(
      <PaginationSelect searchkit={this.searchkit} />
    )
    expect(this.wrapper).toMatchSnapshot()
  })

});
