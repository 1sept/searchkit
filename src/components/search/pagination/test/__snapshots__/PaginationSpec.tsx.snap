// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination tests PaginationSelect 1`] = `
<div
  className="sk-pagination-select is-numbered"
>
  <div
    className="sk-select"
  >
    <select
      onChange={[Function]}
      value={1}
    >
      <option
        disabled={undefined}
        value={1}
      >
        1
      </option>
      <option
        disabled={undefined}
        value={2}
      >
        2
      </option>
      <option
        disabled={undefined}
        value={3}
      >
        3
      </option>
      <option
        disabled={undefined}
        value={4}
      >
        4
      </option>
    </select>
  </div>
</div>
`;

exports[`Pagination tests rendering both disabled on only one total page 1`] = `
<div
  className="sk-pagination-navigation is-numbered"
>
  <div
    className="sk-toggle is-disabled"
    data-qa="options"
  >
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="previous"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Previous Page
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-active"
      data-key={1}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        1
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="next"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Next
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination tests rendering handles pageScope prop 1`] = `
<div
  className="sk-pagination-navigation is-numbered"
>
  <div
    className="sk-toggle"
    data-qa="options"
  >
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="previous"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Previous Page
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-active"
      data-key={1}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        1
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={2}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        2
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="ellipsis-3"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        ...
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key="next"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Next
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination tests rendering handles showNumbers prop 1`] = `
<div
  className="sk-pagination-navigation"
>
  <div
    className="sk-toggle"
    data-qa="options"
  >
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="previous"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Previous Page
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key="next"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Next
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination tests rendering renders eighth page options 1`] = `
<div
  className="sk-pagination-navigation is-numbered"
>
  <div
    className="sk-toggle"
    data-qa="options"
  >
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key="previous"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Previous Page
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={1}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        1
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="ellipsis-2"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        ...
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={5}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        5
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={6}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        6
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={7}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        7
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-active"
      data-key={8}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        8
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="next"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Next
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination tests rendering renders first page options 1`] = `
<div
  className="sk-pagination-navigation is-numbered"
>
  <div
    className="sk-toggle"
    data-qa="options"
  >
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="previous"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Previous Page
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-active"
      data-key={1}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        1
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={2}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        2
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={3}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        3
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={4}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        4
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="ellipsis-5"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        ...
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key="next"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Next
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination tests rendering renders no pagination on no results 1`] = `null`;

exports[`Pagination tests rendering renders second page options 1`] = `
<div
  className="sk-pagination-navigation is-numbered"
>
  <div
    className="sk-toggle"
    data-qa="options"
  >
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key="previous"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Previous Page
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={1}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        1
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-active"
      data-key={2}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        2
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={3}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        3
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={4}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        4
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={5}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        5
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="ellipsis-6"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        ...
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key="next"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Next
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination tests rendering renders text 1`] = `
<div
  className="sk-pagination-navigation is-numbered"
>
  <div
    className="sk-toggle"
    data-qa="options"
  >
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="previous"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Previous Page
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-active"
      data-key={1}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        1
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={2}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        2
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={3}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        3
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key={4}
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        4
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="ellipsis-5"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        ...
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key="next"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Next
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination tests rendering renders without pages 1`] = `
<div
  className="sk-pagination-navigation"
>
  <div
    className="sk-toggle"
    data-qa="options"
  >
    <div
      className="sk-toggle-option sk-toggle__item is-disabled"
      data-key="previous"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Previous Page
      </div>
    </div>
    <div
      className="sk-toggle-option sk-toggle__item"
      data-key="next"
      data-qa="option"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="sk-toggle-option__text"
        data-qa="label"
      >
        Next
      </div>
    </div>
  </div>
</div>
`;
